sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/BaseHibernatePaginatedDataSource.java

```diff
@@ -115,10 +110,30 @@ default Criteria createPaginationCriteria() throws IOException {
    */
   Criterion searchRestrictions(String query);
 
-  /**
-   * Add additional restrictions to the {@link Criteria} object. Note, during listing, a collection of IDs is fetched and then these are
-   * used to perform a second query. In that instance, this method is <b>not</b> called for the ID-driven fetch.
-   */
-  default void setAdditionalPaginationCriteria(Filter filter, Criteria criteria) {
+  @Override
+  default void setFulfilled(Criteria item, boolean isFulfilled) {
+    throw new IllegalArgumentException();
```

Comment:
I didn't really see the problem with your previous design of filters, but this seems worse to me. This filter only makes sense for orders, but could be set on anything and will be a runtime error. This will get much more confusing when we add advanced search and many more fields need to be filterable